src/main/java/com/pusher/client/channel/impl/PrivateChannelImpl.java

```diff
@@ -53,67 +52,46 @@ public void trigger(final String eventName, final String data) {
         }
 
-        try {
-
-            final Map<Object, Object> jsonPayload = new LinkedHashMap<Object, Object>();
-            jsonPayload.put("event", eventName);
-            jsonPayload.put("channel", name);
-            jsonPayload.put("data", data);
-
-            final String jsonMessage = GSON.toJson(jsonPayload);
-            connection.sendMessage(jsonMessage);
-
-        }
-        catch (final JsonSyntaxException e) {
-            throw new IllegalArgumentException("Cannot trigger event " + eventName + " because \"" + data
-                    + "\" could not be parsed as valid JSON");
-        }
+        connection.sendMessage(
+                GSON.toJson(
+                        new TriggerMessage(eventName, name, data)));
     }
 
     /* Base class overrides */
 
     @Override
     public void bind(final String eventName, final SubscriptionEventListener listener) {
 
-        if (listener instanceof PrivateChannelEventListener == false) {
+        if (!(listener instanceof PrivateChannelEventListener)) {
             throw new IllegalArgumentException(
                     "Only instances of PrivateChannelEventListener can be bound to a private channel");
         }
 
         super.bind(eventName, listener);
     }
 
-    @Override
-    @SuppressWarnings("rawtypes")
-    public String toSubscribeMessage() {
-
-        final String authResponse = getAuthResponse();
-
+    private String authenticate() {
         try {
-            final Map authResponseMap = GSON.fromJson(authResponse, Map.class);
-            final String authKey = (String)authResponseMap.get("auth");
-            channelData = (String)authResponseMap.get("channel_data");
-
-            final Map<Object, Object> jsonObject = new LinkedHashMap<Object, Object>();
-            jsonObject.put("event", "pusher:subscribe");
-
-            final Map<Object, Object> dataMap = new LinkedHashMap<Object, Object>();
-            dataMap.put("channel", name);
-            dataMap.put("auth", authKey);
-            if (channelData != null) {
-                dataMap.put("channel_data", channelData);
+            final AuthResponse authResponse = GSON.fromJson(getAuthResponse(), AuthResponse.class);
+            channelData = (String) authResponse.getChannelData();
+
+            if (authResponse.getAuth() == null) {
+                throw new AuthorizationFailureException("Didn't receive all the fields expected " +
+                        "from the Authorizer, expected an auth and shared_secret.");
+            } else {
+                return authResponse.getAuth();
             }
-
-            jsonObject.put("data", dataMap);
-
-            final String json = GSON.toJson(jsonObject);
-            return json;
-        }
-        catch (final Exception e) {
-            throw new AuthorizationFailureException("Unable to parse response from Authorizer: " + authResponse, e);
+        }  catch (JsonSyntaxException e) {
```

Comment:
What are the `JsonSyntaxException` error messages like? It could be helpful to append this to the error message in the rethrown exception.